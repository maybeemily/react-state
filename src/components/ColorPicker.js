import React from 'react';
import PropTypes from 'prop-types';

function ColorPicker({ colorArray, selectColor }) {

  const buttons = colorArray.map(color => {
    return (
      <button key={color} style={{ background: color, height: '75px', width: '75px' }} onClick={selectColor.bind(null, color)}> {color}</button>
    );
  });

  return (
      <>
      { buttons }
      </>
  );
}

ColorPicker.propTypes = {
  colorArray: PropTypes.array.isRequired,
  selectColor: PropTypes.func.isRequired
};

export default ColorPicker;

