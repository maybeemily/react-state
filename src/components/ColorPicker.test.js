import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('Color Picker tests', () => {
  it('renders the color picker', () => {
    const colorArray = ['red', 'blue', 'yellow'];
    const wrapper = shallow(<ColorPicker colors = { colorArray }/>);
    expect(wrapper).toMatchSnapshot();
  });
});
