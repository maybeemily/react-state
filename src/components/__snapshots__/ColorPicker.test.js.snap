// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Picker tests renders the color picker 1`] = `
<Fragment>
  <button
    key="red"
    onClick={[Function]}
    style={
      Object {
        "background": "red",
        "height": "100px",
        "width": "100px",
      }
    }
  >
    red
  </button>
  <button
    key="blue"
    onClick={[Function]}
    style={
      Object {
        "background": "blue",
        "height": "100px",
        "width": "100px",
      }
    }
  >
    blue
  </button>
  <button
    key="yellow"
    onClick={[Function]}
    style={
      Object {
        "background": "yellow",
        "height": "100px",
        "width": "100px",
      }
    }
  >
    yellow
  </button>
  <div
    style={
      Object {
        "background": "grey",
        "height": "100px",
        "width": "100px",
      }
    }
  >
    your chosen color
  </div>
</Fragment>
`;
